packages/contracts-bedrock/scripts/deploy/Deploy.s.sol

```diff
@@ -747,9 +750,53 @@ contract Deploy is Deployer {
                 splitDepth: cfg.faultGameSplitDepth(),
                 clockExtension: Duration.wrap(uint64(cfg.faultGameClockExtension())),
                 maxClockDuration: Duration.wrap(uint64(cfg.faultGameMaxClockDuration())),
-                vm: IBigStepper(
-                    new AlphabetVM(outputAbsolutePrestate, IPreimageOracle(artifacts.mustGetAddress("PreimageOracle")))
-                ),
+                vm: IBigStepper(artifacts.mustGetAddress("MipsSingleton")),
+                weth: weth,
+                anchorStateRegistry: IAnchorStateRegistry(artifacts.mustGetAddress("AnchorStateRegistryProxy")),
+                l2ChainId: cfg.l2ChainID()
+            })
+        });
+    }
+
+    /// @notice Sets the implementation for the `PERMISSIONED_SUPER_CANNON` game type in the `DisputeGameFactory`
+    function setSuperPermissionedGameImplementation() public onlyDevnet broadcast {
+        console.log("Setting SuperPermissionedDisputeGame implementation");
+        IDisputeGameFactory factory = IDisputeGameFactory(artifacts.mustGetAddress("DisputeGameFactoryProxy"));
+        IDelayedWETH weth = IDelayedWETH(artifacts.mustGetAddress("DelayedWETHProxy"));
+
+        _setFaultGameImplementation({
+            _factory: factory,
+            _params: IFaultDisputeGame.GameConstructorParams({
+                gameType: GameTypes.SUPER_CANNON,
+                absolutePrestate: loadMipsAbsolutePrestate(),
+                maxGameDepth: cfg.faultGameMaxDepth(),
+                splitDepth: cfg.faultGameSplitDepth(),
+                clockExtension: Duration.wrap(uint64(cfg.faultGameClockExtension())),
+                maxClockDuration: Duration.wrap(uint64(cfg.faultGameMaxClockDuration())),
+                vm: IBigStepper(artifacts.mustGetAddress("MipsSingleton")),
+                weth: weth,
+                anchorStateRegistry: IAnchorStateRegistry(artifacts.mustGetAddress("AnchorStateRegistryProxy")),
+                l2ChainId: cfg.l2ChainID()
+            })
+        });
+    }
+
+    /// @notice Sets the implementation for the `SUPER_CANNON` game type in the `DisputeGameFactory`
+    function setSuperFaultGameImplementation() public onlyDevnet broadcast {
+        console.log("Setting SuperFaultDisputeGame implementation");
+        IDisputeGameFactory factory = IDisputeGameFactory(artifacts.mustGetAddress("DisputeGameFactoryProxy"));
+        IDelayedWETH weth = IDelayedWETH(artifacts.mustGetAddress("DelayedWETHProxy"));
+
+        _setFaultGameImplementation({
+            _factory: factory,
+            _params: IFaultDisputeGame.GameConstructorParams({
+                gameType: GameTypes.SUPER_CANNON,
+                absolutePrestate: loadMipsAbsolutePrestate(),
```

Comment:
same deal here. Use the interop prestate instead.

Reply:
Where would I pull that from?

Reply:
The interop prestate can be found at the monorepo root in `./op-program/bin/prestate-proof-interop.json`.

As an example, [`_loadDevnetMtMipsAbsolutePrestate`](https://github.com/ethereum-optimism/optimism/blob/669a1fed6ad5f5850d4d55a8f68849a87e988219/packages/contracts-bedrock/scripts/deploy/Deploy.s.sol#L696) used by the `loadMipsAbsolutePrestate` function loads the multi-threaded pre-interop prestate from a similar location. The prestate schema is identical, so you could write a similar function to extract the prestate hash.